src/app/youtube/services/search.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core'
+import { HttpClient } from '@angular/common/http'
+import { Subject } from 'rxjs'
+import { IVideosResponse } from '../models/videos-response.model'
+import { IVideoItem } from '../models/video-item.model'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SearchService {
+  private url = './assets/response.json'
+
+  response?: IVideosResponse
+
+  videos: IVideoItem[] = []
+
+  videos$ = new Subject<IVideoItem[]>()
+
+  videoDetails$ = new Subject<IVideoItem | undefined>()
+
+  constructor(private http: HttpClient) {
+    this.getVideos()
+  }
+
+  getVideos() {
+    this.http.get<IVideosResponse>(this.url).subscribe(response => {
+      this.response = response
+      this.videos = response.items
+      this.videos$.next(this.videos)
+    })
+  }
+
+  getById(id: string) {
+    this.http.get<IVideosResponse>(this.url).subscribe(response => {
+      this.response = response
+      const video = response.items.find(v => v.id === id)
+      this.videoDetails$.next(video)
```

Comment:
it is good to separate service logic to layers. You can google something like "angular service layers facades / angular service architecture / etc..". (https://indepth.dev/posts/1232/designing-scalable-angular-applications)

I have a small example here how you can divide your logic between services (https://github.com/fomenkogregory/angular-examples/tree/service-layers/src/modules/cars)

store service just stores the data in subjects
http service does http calls (and only that)
facade is using store and http and preparing data and methods for usage in component

Reply:
this comment is just for knowledge, you don't have to fix this right away